base.go

```diff
@@ -172,18 +172,26 @@ func InitDefaults(args *ReleaseType, chart, repo string, values interface{}) {
 	// Most strongly typed charts will have a default chart name as well as a default
 	// repository location. If available, set those. The user might override these,
 	// so only initialize them if they're empty.
-	if args.Chart == "" {
-		args.Chart = chart
+	if args.Chart == nil {
+		args.Chart = pulumi.String(chart)
 	}
 	if args.RepositoryOpts.Repo == nil {
-		args.RepositoryOpts.Repo = &repo
+		args.RepositoryOpts.Repo = toStringPtr(&repo)
 	}
 
 	// Blit the strongly typed values onto the weakly typed values, so that the Helm
 	// Release is constructed properly. In the event a value is present in both, the
 	// strongly typed values override the weakly typed map.
+	localMap := make(map[string]interface{})
 	if args.Values == nil {
-		args.Values = make(map[string]interface{})
+		args.Values = pulumi.ToMap(localMap)
+	} else {
+		args.Values.ToMapOutput().ApplyT(func(vals map[string]interface{}) map[string]interface{} {
```

Comment:
```suggestion
		merged := args.Values.ApplyT(func(vals map[string]interface{}) map[string]interface{} {
```
And then on line 194 `args.Values = merged`?

I haven't checked the types and perhaps some conversions would be required...